### PR TITLE
refactor get_address and wait_for_login functions

### PR DESCRIPTION
**Pull request by xutian**
Add step to try to login guest via serial console and update VM address cache if wait for get VM address  failed, so that we can avoid some unexpected  test error.

### CODE REVIEW

**Comment by xutian** [comment]
Still in testing, will update test results when job done.

**Comment by sathnaga** [comment]
@xutian looks good to me, will test with my multi interface tests and update

**Comment by sathnaga** [comment]
@xutian works fine with flexible_nic_index  param 
 ```
(1/4) guest_sanity.boundary.network_max.spaprvlan.qemu.qcow2.virtio_scsi.smp2.spapr-vlan.Fedora.25.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (166.79 s)
 (2/4) guest_sanity.boundary.network_max.spaprvlan.qemu.qcow2.virtio_scsi.smp2.spapr-vlan.Fedora.25.ppc64le.powerkvm-libvirt.remove_guest.without_disk: PASS (3.38 s)
 (3/4) guest_sanity.boundary.network_max.virtionet.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (83.38 s)
 (4/4) guest_sanity.boundary.network_max.virtionet.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.remove_guest.without_disk: PASS (3.18 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
TESTS TIME : 256.73 s

```

**Comment by xutian** [comment]
@sathnaga  thanks your test results.

**Comment by sathnaga** [comment]
@xutian wc :-)

**Comment by yanan-fu** [comment]
Run one acceptance test loop,  no login timeout issue that caused by wrong ip address.

Acked-by: Yanan Fu yfu@redhat.com

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3335,7 +3336,9 @@ def update_mac_ip_address(vm, params, timeout=None):
                                                restart_network=True)
             addr_map = get_guest_address_map(session)
             session.close()
-        vm.address_cache.update(addr_map)
+        if addr_map:
+            logging.debug("Update address_cache: %s" % addr_map)
```
Comment (balamuruhans) [inline virttest/utils_net.py]: `logging.debug("Update address_cache: %s", addr_map)`

File: virttest/virt_vm.py [hunk 2]
```diff
@@ -655,106 +658,92 @@ def get_address(self, index=0):
...
                 else:
-                    raise VMAddressVerificationError(nic.mac, arp_ip)
-
+                    if mac in self.address_cache:
+                        self.address_cache[mac] = None
+                    else:
+                        # clean item in host arp table
+                        os.system("arp -d %s" % ip_addr)
+                    logging.debug("Clean up outdated address info, "
+                                  "MAC: %s, IP: %s" % (mac, ip_addr))
```
Comment (balamuruhans) [inline virttest/virt_vm.py]: same here

**Comment by xutian** [comment]
Merged, thanks  @yanan-fu  and @sathnaga

**Pull request merged**